app/src/main/kotlin/compiler/syntax/AstFactory.kt

```diff
@@ -106,6 +106,63 @@ class AstFactory(private val diagnostics: Diagnostics) {
         return rotateExpressionLeft(newTopLevelNode)
     }
 
+    fun makeArrayLoop(parseTree: ParseTree<Symbol>): Statement {
```

Comment:
I meant for this function to take AST nodes as parameters.

Reply:
I think that this is also fine. I will have to refactor it a little bit merging anyway.

Reply:
Which ast nodes? `arrayExpression`, `receivingVariable` and `bodyBlock`?

Reply:
let's go